src/commands/createFunction/createFunction.ts

```diff
@@ -49,20 +48,24 @@ export async function createFunctionFromCommand(
 export async function createFunctionInternal(context: IActionContext, options: api.ICreateFunctionOptions): Promise<void> {
     let workspaceFolder: WorkspaceFolder | undefined;
     let workspacePath: string | undefined = options.folderPath;
+
     if (workspacePath === undefined) {
-        workspaceFolder = await getWorkspaceFolder(context, options);
-        workspacePath = workspaceFolder.uri.fsPath;
+        workspaceFolder = await getWorkspaceFolder();
+        workspacePath = workspaceFolder?.uri.fsPath;
     } else {
         workspaceFolder = getContainingWorkspace(workspacePath);
     }
 
-    addLocalFuncTelemetry(context, workspacePath);
-
-    const projectPath: string | undefined = await verifyAndPromptToCreateProject(context, workspaceFolder || workspacePath, options);
+    let projectPath: string | undefined = await verifyProjectPath(context, workspaceFolder || workspacePath);
     if (!projectPath) {
+        // If we cannot find a valid Functions project, we need to put the user into the 'Create New Project' flow..
+        context.telemetry.properties.noWorkspaceResult = 'createNewProject';
+        await createNewProjectInternal(context, options);
         return;
     }
 
+    addLocalFuncTelemetry(context, workspacePath);
```

Comment:
I think you should move this back up to where it was to capture telemetry.

Reply:
Because of the change I've made in `getWorkspaceFolder` workspacePath may sometimes be `undefined` which is why I moved it down to a spot where I know `workspacePath` will always be defined.  Does this matter for our purposes in telemetry or is `undefined` an okay value to record?

Reply:
Actually, don't worry about it. It's already done in `createNewProjectInternal`. But to answer your question, yes, it's fine as the function definition is as follows:

`export function addLocalFuncTelemetry(context: IActionContext, workspacePath: string | undefined)`

Reply:
I ended up moving it back before `createNewProjectInternal`.  Let me know if you prefer it back down after `createNewProjectInternal`.